BGL/include/CGAL/boost/graph/helpers.h

```diff
@@ -355,29 +356,210 @@ bool is_valid_face_descriptor( typename boost::graph_traits<FaceGraph>::face_des
   return true;
 }
 
-
-template <typename FaceGraph>
-bool is_valid_polygon_mesh(const FaceGraph& g)
+/*!
+  \ingroup PkgBGLHelperFct
+ * \brief is_valid_polygon_mesh checks the integrity of a `PolygonMesh`.
+ * \param g the `PolygonMesh` to test.
+ * \param verb : if `true`, the details of the check will be written in the standard output.
+ *
+ * \tparam PolygonMesh a model of `FaceListGraph` and `HalfedgeListGraph`
+ * \return `true` if the `PolygonMesh` is valid, `false` otherwise.
+ */
+template <typename PolygonMesh>
+bool is_valid_polygon_mesh(const PolygonMesh& g, bool verb = false)
 {
-  typedef typename boost::graph_traits<FaceGraph>::halfedge_descriptor halfedge_descriptor;
-  typedef typename boost::graph_traits<FaceGraph>::vertex_descriptor   vertex_descriptor;
-  typedef typename boost::graph_traits<FaceGraph>::face_descriptor     face_descriptor;
-  BOOST_FOREACH(vertex_descriptor v, vertices(g)){
-    if(! is_valid_vertex_descriptor(v,g)){
-      return false;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedge_descriptor   halfedge_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_descriptor     vertex_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::face_descriptor       face_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_iterator       vertex_const_iterator;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedge_iterator     halfedge_const_iterator;
+  typedef typename boost::graph_traits<PolygonMesh>::face_iterator         face_const_iterator;
+  typedef typename boost::graph_traits<PolygonMesh>::vertices_size_type    vertex_size_type;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedges_size_type   halfedges_size_type;
+  typedef typename boost::graph_traits<PolygonMesh>::faces_size_type       faces_size_type;
```

Comment:
The [testsuite is a lot yellow, today](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.12-Ic-175/Convex_decomposition_3/TestReport_gimeno_Debian-Stable.gz):
```
In file included from /mnt/testsuite/include/CGAL/Nef_3/polygon_mesh_to_nef_3.h:35:0,
                 from /mnt/testsuite/include/CGAL/Nef_polyhedron_3.h:57,
                 from /home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Convex_decomposition_3/check_decomposition.cpp:3:
/mnt/testsuite/include/CGAL/boost/graph/helpers.h: In function 'bool CGAL::is_valid_polygon_mesh(const PolygonMesh&, bool)':
/mnt/testsuite/include/CGAL/boost/graph/helpers.h:374:76: warning: typedef 'vertex_const_iterator' locally defined but not used [-Wunused-local-typedefs]
   typedef typename boost::graph_traits<PolygonMesh>::vertex_iterator       vertex_const_iterator;
                                                                            ^~~~~~~~~~~~~~~~~~~~~
/mnt/testsuite/include/CGAL/boost/graph/helpers.h:375:76: warning: typedef 'halfedge_const_iterator' locally defined but not used [-Wunused-local-typedefs]
   typedef typename boost::graph_traits<PolygonMesh>::halfedge_iterator     halfedge_const_iterator;
                                                                            ^~~~~~~~~~~~~~~~~~~~~~~
/mnt/testsuite/include/CGAL/boost/graph/helpers.h:376:76: warning: typedef 'face_const_iterator' locally defined but not used [-Wunused-local-typedefs]
   typedef typename boost::graph_traits<PolygonMesh>::face_iterator         face_const_iterator;
                                                                            ^~~~~~~~~~~~~~~~~~~
```